deltacat/tests/compute/test_compact_partition_rebase_then_incremental.py

```diff
@@ -69,239 +76,183 @@ def setup_compaction_artifacts_s3_bucket(setup_s3_resource: ServiceResource):
+            validation_callback_func_kwargs,
+            create_table_strategy,
+            incremental_deltas_arrow_arrays_param,
+            incremental_deltas_delta_type,
+            rebase_expected_compact_partition_result,
             compact_partition_func,
-        ) in INCREMENTAL_TEST_CASES.items()
+        ) in REBASE_THEN_INCREMENTAL_TEST_CASES.items()
     ],
-    ids=[test_name for test_name in INCREMENTAL_TEST_CASES],
+    ids=[test_name for test_name in REBASE_THEN_INCREMENTAL_TEST_CASES],
     indirect=[],
 )
-def test_compact_partition_incremental(
+def test_compact_partition_rebase_then_incremental(
     request: pytest.FixtureRequest,
     setup_s3_resource: ServiceResource,
     setup_local_deltacat_storage_conn: Dict[str, Any],
-    setup_compaction_artifacts_s3_bucket: None,
     test_name: str,
-    source_table_version: str,
-    destination_table_version: str,
     primary_keys_param: Set[str],
-    sort_keys_param,
-    partition_keys_param,
+    sort_keys_param: Dict[str, str],
+    partition_keys_param: Dict[str, str],
+    partition_values_param: str,
     column_names_param: List[str],
-    arrow_arrays_param: List[pa.Array],
-    rebase_source_partition_locator_param,
-    partition_values_param,
-    expected_result,
-    validation_callback_func,  # use and implement func and func_kwargs if you want to run additional validations apart from the ones in the test
-    validation_callback_func_kwargs,
-    do_teardown_local_deltacat_storage_db,
-    use_prev_compacted,
-    create_placement_group_param,
-    records_per_compacted_file_param,
-    hash_bucket_count_param,
-    compact_partition_func,
+    input_deltas_arrow_arrays_param: Dict[str, pa.Array],
+    input_deltas_delta_type: str,
+    expected_terminal_compact_partition_result: pa.Table,
+    create_placement_group_param: bool,
+    records_per_compacted_file_param: int,
+    hash_bucket_count_param: int,
+    validation_callback_func: Callable,  # use and implement func and func_kwargs if you want to run additional validations apart from the ones in the test
+    validation_callback_func_kwargs: Dict[str, Any],
+    create_table_strategy: Callable,
+    incremental_deltas_arrow_arrays_param: Dict[str, pa.Array],
+    incremental_deltas_delta_type: str,
+    rebase_expected_compact_partition_result: pa.Table,
+    compact_partition_func: Callable,
 ):
     import deltacat.tests.local_deltacat_storage as ds
     from deltacat.types.media import ContentType
     from deltacat.storage import (
+        DeltaType,
+        Partition,
         PartitionLocator,
+        Stream,
     )
     from deltacat.compute.compactor.model.compact_partition_params import (
         CompactPartitionParams,
     )
     from deltacat.utils.placement import (
         PlacementGroupManager,
     )
-    from deltacat.compute.compactor import (
-        RoundCompletionInfo,
-    )
 
     ds_mock_kwargs = setup_local_deltacat_storage_conn
+    ray.shutdown()
```

Comment:
Why does ray need to be shut down and switched to local? Can this be handled during test setup instead?

Reply:
Removing the `ray.shutdown()` invocation. We can just use ignore_reinit_error=True argument. There seems to be issues with the unit tests (most of the `compact_partition` test cases are failing) in the `CI` workflow if we handle it as a module-level fixture that seems to be idiosyncratic to the GitHub Actions runner but I haven't yet spent time investigating it. I'll look into it

Reply:
Now fixed - problem was with placement group creation interacting with a resource constrained Github Actions runner.  See https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources